docs/reference/persistence-configuration.txt

```diff
@@ -177,22 +188,52 @@ failing subsequent updates:
   band.with(collection: "artists") do |band_object|
     band_object.save!
   end
-  
+
   # This will not save - updates the collection "bands" which does not have
   # the Scuba band
   band.update_attribute(likes: 1000)
-  
+
   # This will update the document.
   band.with(collection: "artists") do |band_object|
     band_object.update_attribute(likes: 1000)
   end
 
-As of Mongoid 6.0, the ``with`` method must always be called with a block,
-and the temporary persistence context exists only for the duration of the block.
-This is because a new client is created under the covers with the options
-passed to ``with``. To ensure that this client is closed and its associated
-resources are freed, the scope when this client could be used must be
-well-defined.
+The ``with`` command can also be used without a block:
+
+.. code-block:: ruby
+
+  # The #with method on Model classes returns a criteria.
+  criteria = Band.with(database: "music-non-stop")
+  # => #<Mongoid::Criteria
+         selector: {}
+         options:  {}
+         class:    Band
+         embedded: false>
+
+  criteria.first
+  criteria.clear_persistance_context!
+
+  # The #with method on documents modifies the document object itself since
+  # calling #dup on it will change that document's _id. The #with method on
+  # documents will also return the same document.
+  band = Band.new
+  band.with(client: :tertiary)
```

Comment:
This line mutates `band` permanently? I understand how this came to be (`band` would be mutated in the block previously) but this permanent mutation feels too iffy to me.

Reply:
This looks weird to me, too. But I think this is an inevitable side effect of the feature.

Reply:
This feature should not cause any permanent mutation of globals. That will be a disaster, not to mention not thread-safe, etc. It should return a new criteria object where the modification happens specific to that criteria.

Reply:
Note that calling clear_persistence_context does put it back to its original context still...

Reply:
For the model instances which are mutated by `with`, my suggestion was to define `with!` and in the `with` implementation require a block. For model classes retain the current behavior of `with` only which returns a new scope.

Reply:
I think `with!` is too dangerous to be practical in real-world applications. It would be better to declare any such options in model itself (declarative > imperitive)

Reply:
done